File: php/features.php
<?php
declare(strict_types=1);

// SerialifColor SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SerialifColorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SerialifColorBaseFeature();
            case "test":
                return new SerialifColorTestFeature();
            default:
                return new SerialifColorBaseFeature();
        }
    }
}

File: py/features.py
# SerialifColor SDK feature factory

from feature.base_feature import SerialifColorBaseFeature
from feature.test_feature import SerialifColorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SerialifColorBaseFeature(),
        "test": lambda: SerialifColorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
